shell/components/nav/TopLevelMenu.vue

```diff
@@ -259,6 +259,40 @@ export default {
     productFromRoute() {
       return getProductFromRoute(this.$route);
     },
+
+    appBar() {
+      let activeFound = false;
+
+      const appBar = {
+        hciApps:           this.hciApps,
+        multiClusterApps:  this.multiClusterApps,
+        legacyApps:        this.legacyApps,
+        configurationApps: this.configurationApps,
+        pinFiltered:       this.pinFiltered,
+        clustersFiltered:  this.clustersFiltered,
+      };
+
+      Object.keys(appBar).forEach((menuSection) => {
+        const menuSectionItems = appBar[menuSection];
+        const isClusterCheck = menuSection === 'pinFiltered' || menuSection === 'clustersFiltered';
+
+        // need to reset active state on other menu items
+        menuSectionItems.forEach((item) => {
+          item.isMenuActive = false;
+        });
+
+        if (!activeFound) {
+          const currActiveMenu = menuSectionItems.find((item) => this.checkActiveRoute(item, isClusterCheck));
+
+          if (currActiveMenu) {
+            activeFound = true;
+            currActiveMenu.isMenuActive = true;
+          }
+        }
```

Comment:
I think that performing the `find` might be a redundant since we're already iterating over `menuSectionItems` to set the active state to false

```suggestion
        menuSectionItems.forEach((item) => {
          item.isMenuActive = false;
          
          if (!activeFound && this.checkActiveRoute(item, isClusterCheck)) {
            activeFound = true;
            item.isMenuActive = true;
          }
        });
```